### PR TITLE
GREEN-20812: add demographic question set API

### PR DESCRIPTION
**Pull request by capablemonkey**

### CODE REVIEW

File: source/includes/harvest/_demographic_data.md [hunk 1]
```diff
@@ -21,8 +44,10 @@ Demographic questions and answers submitted during the application process. For
 | Attribute | Description |
 |-----------|-------------|
 | id | The demographic question's unique identifier |
-| translations.language | Only `en` (English) is supported at this time.
-| translations.name | The question text
+| question_set_id | The demographic question set that this belongs to
+| name | The question text
+| translations.language | Translations have been deprecated.  Only `en` (English) is supported at this time.
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: Update to "Translations have been deprecated but are kept for backwards compatibility."

File: source/includes/harvest/_demographic_data.md [hunk 2]
```diff
@@ -21,8 +44,10 @@ Demographic questions and answers submitted during the application process. For
...
+| translations.name | Translations have been deprecated. This value will be the same as `name` above.
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: Update to "Translations have been deprecated but are kept for backwards compatibility."

File: source/includes/harvest/_demographic_data.md [hunk 3]
```diff
@@ -47,8 +73,9 @@ Demographic questions and answers submitted during the application process. For
 | Attribute | Description |
 |-----------|-------------|
 | id | The demographic answer option's unique identifier |
-| translations.language | Only `en` (English) is supported at this time.
-| translations.name | The answer option text.
+| name | The answer option text
+| translations.language | Translations have been deprecated.  Only `en` (English) is supported at this time.
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: Update to "Translations have been deprecated but are kept for backwards compatibility."

File: source/includes/harvest/_demographic_data.md [hunk 4]
```diff
@@ -47,8 +73,9 @@ Demographic questions and answers submitted during the application process. For
...
+| translations.name | Translations have been deprecated. This value will be the same as `name` above.
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: Update to "Translations have been deprecated but are kept for backwards compatibility."

File: source/includes/harvest/_demographic_data.md [hunk 5]
```diff
@@ -2,11 +2,34 @@
...
+| title | The title of the demographic question set
+| description | The demographic question set's description.  This is a rich text field which may contain HTML.
+| active | If `false`, the demographic question set has been deleted.
+
 ## The Demographic Question object
 
 ```json
 {
   "id": 123,
+  "question_set_id": 456,
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: Great catch!  Yes, not sure how I missed this the first time around.  It should definitely be named `demographic_question_id` for clarity and consistency's sake.

File: source/includes/harvest/_demographic_data.md [hunk 6]
```diff
@@ -21,8 +44,10 @@ Demographic questions and answers submitted during the application process. For
 | Attribute | Description |
 |-----------|-------------|
 | id | The demographic question's unique identifier |
-| translations.language | Only `en` (English) is supported at this time.
-| translations.name | The question text
+| question_set_id | The demographic question set that this belongs to
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: Sure. Good idea.

[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: Hmm, actually... I'm not sure if it's possible for a demographic question to not belong to a demographic question set.  When creating a new question in the UI, it seems you must add the question to a question set.

File: source/includes/harvest/_demographic_data.md [hunk 7]
```diff
@@ -77,6 +104,49 @@ Demographic questions and answers submitted during the application process. For
...
+`GET https://harvest.greenhouse.io/v1/demographics/question_sets`
+
+### Querystring parameters
+
+| Parameter | Description |
+|-----------|-------------|
+| *per_page | Return up to this number of objects per response. Must be an integer between 1 and 500. Defaults to 100.
+
+<br>
+[See noteworthy response attributes.](#the-demographic-question-object)
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: 👍

File: source/includes/harvest/_demographic_data.md [hunk 8]
```diff
@@ -126,6 +200,40 @@ curl 'https://harvest.greenhouse.io/v1/demographics/questions'
 
 This endpoint supports pagination. See the [Pagination](#pagination) section for more detail.
 
+## GET: Retrieve Demographic Question Set
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: 👍

File: source/includes/harvest/_demographic_data.md [hunk 9]
```diff
@@ -468,4 +583,64 @@ Parameter | Description
 id | ID of the demographic answer you want to retrieve.
 
 <br>
-[See noteworthy response attributes.](#the-demographic-answer-object)
\ No newline at end of file
+[See noteworthy response attributes.](#the-demographic-answer-object)
+
+## GET: List Demographic Questions For Demographic Question Set
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: 👍

File: source/includes/harvest/_demographic_data.md [hunk 10]
```diff
@@ -468,4 +583,64 @@ Parameter | Description
...
+id | ID of the demographic question set for which you want to retrieve demographic questions
+
+### Querystring parameters
+
+| Parameter | Description |
+|-----------|-------------|
+| *per_page | Return up to this number of objects per response. Must be an integer between 1 and 500. Defaults to 100.
+
+<br>
+[See noteworthy response attributes.](#the-demographic-question-set-object)
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: 👍 nice catch

File: source/includes/harvest/_job_posts.md [hunk 11]
```diff
@@ -7,6 +7,7 @@ Describes the online job posts for an organization's jobs (as seen on the Job Bo
 ```json
 {
   "id": 123,
+  "demographic_question_set_id": 999,
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_job_posts.md]: 👍

[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_job_posts.md]: sure

File: source/includes/harvest/_job_posts.md [hunk 12]
```diff
@@ -143,6 +144,7 @@ Describes the online job posts for an organization's jobs (as seen on the Job Bo
 | Attribute | Description |
 |-----------|-------------|
 | id | Job post ID
+| demographic_question_set_id | The demographic question set associated with this job post
```
[the remark this answers is not in the document]
Comment (capablemonkey) [reply source/includes/harvest/_job_posts.md]: 👍

File: source/includes/harvest/_demographic_data.md (hunk 1 above)
Comment (capablemonkey) [reply source/includes/harvest/_demographic_data.md]: 👍

**Comment by capablemonkey** [comment]
Thanks for taking a look @marvin-gh!  Ready for another pass

File: source/includes/harvest/_demographic_data.md [hunk 13]
```diff
@@ -92,6 +197,8 @@ curl 'https://harvest.greenhouse.io/v1/demographics/questions'
 [
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: Also missing `active`

File: source/includes/harvest/_demographic_data.md [hunk 14]
```diff
@@ -126,6 +235,66 @@ curl 'https://harvest.greenhouse.io/v1/demographics/questions'
...
+
+```shell
+curl 'https://harvest.greenhouse.io/v1/demographics/question_sets/{id}/questions'
+-H "Authorization: Basic MGQwMzFkODIyN2VhZmE2MWRjMzc1YTZjMmUwNjdlMjQ6"
+```
+
+> The above command returns JSON structured like this:
+
+```json
+[
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: missing `active` in each of these

File: source/includes/harvest/_demographic_data.md [hunk 15]
```diff
@@ -140,6 +309,8 @@ curl 'https://harvest.greenhouse.io/v1/demographics/questions/{id}'
 ```json
 {
```
Comment (marvin-gh) [inline source/includes/harvest/_demographic_data.md]: also missing `active`

**Comment by capablemonkey** [comment]
> Noticed the `active` flag was historically missing in examples for demographic questions, but approving anyway since that isn't introduced by your changes.

ah, nice catch!  I updated the examples

**Comment by capablemonkey** [comment]
> Noticed the `active` flag was historically missing in examples for demographic questions, but approving anyway since that isn't introduced by your changes.

ah, nice catch!  I updated the examples

File: source/includes/harvest/_demographic_data.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (marvin-gh) [reply source/includes/harvest/_demographic_data.md]: Sounds like you're right that questions will always be in a question set. Stef is confirming that there was a data migration to update any existing questions as well.

**Pull request merged**